File: cad_automator/core_backend/factory.py
from .processors.cleaner import SmartCleaner
from .processors.merger import LayerMerger

class ProcessorFactory:
    """
    [Factory Class]
    简单工厂类，负责对象的实例化。
    前端不需要知道 'SmartCleaner' 怎么创建，只需要请求 'smart_clean' 即可。
    方便未来添加 'BatchPrint', 'FormatConverter' 等新功能。
    """
    @staticmethod
    def get_processor(task_type: str):
        if task_type == "smart_clean":
            return SmartCleaner()
        elif task_type == "layer_merge":
            return LayerMerger()
        else:
            raise ValueError(f"工厂报错: 未知的任务类型 '{task_type}'")

File: cad-processor/src-tauri/src/factory.rs
use super::processor::CadProcessor;
use super::processors::cleaner::SmartCleaner;
use super::processors::merger::LayerMerger;
use super::processors::flattener::Flattener;
use super::processors::exploder::Exploder;

// [Factory]
// 处理器工厂
pub struct ProcessorFactory;

impl ProcessorFactory {
    // 这里也是 "前后端分离" 的关键点
    // 接受一个字符串 key，返回一个动态分发的 Trait 对象 (Box<dyn CadProcessor>)
    pub fn create(action_type: &str) -> Option<Box<dyn CadProcessor>> {
        match action_type {
            "smart_clean" => Some(Box::new(SmartCleaner)),
            "layer_merge" => Some(Box::new(LayerMerger)),
            "flatten" => Some(Box::new(Flattener)),
            "explode" => Some(Box::new(Exploder)),
            _ => None,
        }
    }
}
